src/types/markers.js

```diff
@@ -144,6 +155,17 @@ export type MarkerSchema = {|
 
   // if present, give the marker its own local track
   graphs?: Array<MarkerGraph>,
+
+  // If set to true, markers of this type are assumed to be well-nested with all
+  // other stack-based markers on the same thread. Stack-based markers may
+  // be displayed in a different part of the marker chart than non-stack-based
+  // markers.
```

Comment:
I still wonder if simply ordering markers by startTime would be good enough for getting the pyramid shapes? But I'm not opposed.

Reply:
If I understand correctly, the idea of this field is to have a different section in the marker chart (instead of grouping markers by their categories and then by their names), so marker from different categories + names can be still stacked on top of each other.

I think we had the idea of having a checkbox in the marker chart to convert all the markers into stack based markers, right? So all the makers could be nested instead of only a handful selected ones. That could be something we can explore too, but having some markers behave stack based by default can be useful for the default view.

Reply:
> If I understand correctly, the idea of this field is to have a different section in the marker chart (instead of grouping markers by their categories and then by their names), so marker from different categories + names can be still stacked on top of each other.

That's right.

> I think we had the idea of having a checkbox in the marker chart to convert all the markers into stack based markers, right?

I don't think it would work very well if you have non-stack-based markers in there, like CSS animation markers, IPC messages, network markers, FirstContentfulComposite markers, GCMajor markers etc. I really do think we need to have a distinction because otherwise the pyramid shape will be messed up.